docs/forms.md

```diff
@@ -0,0 +1,138 @@
+# Forms
+
+Forms and validation is quite a complex topic! This just covers a few topics for now.
+
+Our approach for forms is to clone an _editing_ version of the data for use in the form. This allows us to:
+- know if the value of the data has changed (`diff(source, edit)`)
+- reset the form (`edit = source`)
+- emit the diff of changes in a `@save` event (`emit(edit)`) (works nicely with http `PATCH`)
+- handle underlying changes in the source value as we wish
+
+In many places we enrich the data with additional fields,
+and we need to use the original data (available as `__unenriched`) to correctly only detect changes in the
+underlying value, not the derrived data.
```

Comment:
typo: derrived

Reply:
Fixed (plus found a bunch more errors in there)